python/cudf/cudf/core/column/column.py

```diff
@@ -414,14 +414,17 @@ def element_indexing(self, index):
         if val.null_count == 1:
             val = None
         else:
-            val = val.to_array()[0]
+            valid = (
+                cudautils.mask_get.py_func(self.mask_array_view, index)
```

Comment:
This will add a kernel invocation and a device to host. I don't think we want to do this.

Reply:
It was already happening before though.

However I'm now wondering it this is actually needed. Would the selection/`null_count` check above already handle this case?

Reply:
Based on offline discussion, it sounds like this isn't needed. Dropping in PR ( https://github.com/rapidsai/cudf/pull/4912 ) and closing this PR out.